### PR TITLE
registrate sealengines through init function

### PR DESCRIPTION
**Pull request by winsvega**
connects to https://github.com/ethereum/webthree-umbrella/issues/341

### CODE REVIEW

File: test/libethereum/BlockChainInsert.cpp [hunk 1]
```diff
@@ -149,9 +149,12 @@ void syncStateTrie(bytesConstRef _block, OverlayDB const& _dbSource, OverlayDB&
 	}
 }
 
-#if !defined(_WIN32)
+/*
```
Comment (chriseth) [inline test/libethereum/BlockChainInsert.cpp:152]: Unrelated change

File: test/libethereum/BlockchainTestsFiller/Homestead/bcUncleHeaderValiditiyFiller.json [hunk 2]
```diff
@@ -1371,5 +1371,448 @@
                 ]
             }
         ]
+    },
+
+    "incorrectUncleNumber1" : {
```
Comment (chriseth) [inline test/libethereum/BlockchainTestsFiller/Homestead/bcUncleHeaderValiditiyFiller.json:1376]: Unrelated change

File: test/libethereum/BlockChainTests.cpp [hunk 3]
```diff
@@ -565,6 +565,15 @@ void overwriteUncleHeaderForTest(mObject& uncleHeaderObj, TestBlock& uncle, std:
 								 (h256)fromHex("1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347"),
 								 uncleHeader.stateRoot()
 								 );
+
+			if (uncleHeaderObj.count("RelTimestamp"))
```
Comment (chriseth) [inline test/libethereum/BlockChainTests.cpp:569]: Unrelated change

File: test/libethereum/BlockChainInsert.cpp (hunk 1 above)
Comment (winsvega) [inline test/libethereum/BlockChainInsert.cpp:152]: once sealEngines used this way, we could comment out this test.

Comment (chriseth) [inline test/libethereum/BlockChainInsert.cpp:152]: Ok but then please do instead of changing the way it is commented out.
